### PR TITLE
Update go dependencies

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Updates:
- github.com/containerd/containerd v1.6.8
- github.com/prometheus/client_golang v1.13.0
- google.golang.org/grpc v1.49.0
- google.golang.org/protobuf v1.28.1
- cloud.google.com/go/storage v1.27.0
- cloud.google.com/go v0.104.0
- github.com/grpc-ecosystem/grpc-gateway/v2 v2.11.3

## Related Issue(s)

Fixes #

## How to test
- Check pods start
- Open a workspace

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by jeanp413 — APPROVED** [review]

**Review by Pothulapati — APPROVED** [review]
:shipit:

File: components/BUILD.yaml [hunk 1]
```diff
@@ -182,14 +182,15 @@ scripts:
...
+      leeway exec --filter-type go -v -- go mod tidy -compat=1.19
 
       # not all the code is present in leeway.
       # ensure we update everything containing a go.mod file
       for COMPONENT in $(find . -name go.mod); do
         DIRECTORY=$(dirname $COMPONENT)
+        echo "$DIRECTORY"
         pushd $DIRECTORY > /dev/null
-        go mod tidy -compat=1.18
+        go mod tidy -compat=1.19 -v
```
Comment (easyCZ) [inline components/BUILD.yaml:193]: 👍

**Review by easyCZ — APPROVED** [review]

**Review by sagor999 — APPROVED** [review]

**Review by vulkoingim — APPROVED** [review]

**Comment by aledbf** [comment]
friendly ping @csweichel @geropl @Furisto @akosyakov

**Review by csweichel — APPROVED** [review]

**Review by Furisto — APPROVED** [review]

**Pull request merged**